frontend/src/metabase-lib/display.ts

```diff
@@ -0,0 +1,168 @@
+import * as Lib from "metabase-lib";
+import type {
+  CardDisplayType,
+  VisualizationSettings,
+} from "metabase-types/api";
+import { TYPE } from "metabase-lib/types/constants";
+import { isa } from "metabase-lib/types/utils/isa";
+
+type DefaultDisplay = {
+  display: CardDisplayType;
+  settings?: Partial<VisualizationSettings>;
+};
+
+export const defaultDisplay = (query: Lib.Query): DefaultDisplay => {
+  const { isNative } = Lib.queryDisplayInfo(query);
+
+  if (isNative) {
+    return { display: "table" };
+  }
+
+  const stageIndex = -1;
+  const aggregations = Lib.aggregations(query, stageIndex);
+  const breakouts = Lib.breakouts(query, stageIndex);
+
+  if (aggregations.length === 0 && breakouts.length === 0) {
+    return { display: "table" };
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 0) {
+    return { display: "scalar" };
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 1) {
+    const [{ columnInfo }] = getBreakoutsWithColumnInfo(
+      query,
+      stageIndex,
+      breakouts,
+    );
+
+    if (isState(columnInfo)) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "region",
+          "map.region": "us_states",
+        },
+      };
+    }
+
+    if (isCountry(columnInfo)) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "region",
+          "map.region": "world_countries",
+        },
+      };
+    }
+  }
+
+  if (aggregations.length >= 1 && breakouts.length === 1) {
+    const [{ breakout, columnInfo }] = getBreakoutsWithColumnInfo(
+      query,
+      stageIndex,
+      breakouts,
+    );
+
+    if (isDate(columnInfo)) {
+      const info = Lib.displayInfo(query, stageIndex, breakout);
+
+      if (info.isTemporalExtraction) {
```

Comment:
```suggestion
      if (info.isTemporalExtraction()) {
```

?

Reply:
It's a boolean attribute, not a function.